qiskit/chemistry/algorithms/excited_states_solvers/qeom.py

```diff
@@ -0,0 +1,523 @@
+
+        # these matrices are numpy arrays and therefore have the ``shape`` attribute
+        # pylint: disable=unsubscriptable-object
+        q_mat = q_mat + q_mat.T - np.identity(q_mat.shape[0]) * q_mat
+        w_mat = w_mat + w_mat.T - np.identity(w_mat.shape[0]) * w_mat
+        m_mat = m_mat + m_mat.T - np.identity(m_mat.shape[0]) * m_mat
+        v_mat = v_mat + v_mat.T - np.identity(v_mat.shape[0]) * v_mat
+
+        q_mat = np.real(q_mat)
+        w_mat = np.real(w_mat)
+        m_mat = np.real(m_mat)
+        v_mat = np.real(v_mat)
+
+        q_mat_std = q_mat_std / float(size**2)
+        w_mat_std = w_mat_std / float(size**2)
+        m_mat_std = m_mat_std / float(size**2)
+        v_mat_std = v_mat_std / float(size**2)
+
+        logger.debug("\nQ:=========================\n%s", q_mat)
+        logger.debug("\nW:=========================\n%s", w_mat)
+        logger.debug("\nM:=========================\n%s", m_mat)
+        logger.debug("\nV:=========================\n%s", v_mat)
+
+        return m_mat, v_mat, q_mat, w_mat, m_mat_std, v_mat_std, q_mat_std, w_mat_std
+
+    @staticmethod
+    def _compute_excitation_energies(m_mat: np.ndarray, v_mat: np.ndarray, q_mat: np.ndarray,
+                                     w_mat: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
+        """Diagonalizing M, V, Q, W matrices for excitation energies.
+
+        Args:
+            m_mat : M matrices
+            v_mat : V matrices
+            q_mat : Q matrices
+            w_mat : W matrices
+
+        Returns:
+            1-D vector stores all energy gap to reference state
+            2-D array storing the X and Y expansion coefficients
+        """
+        logger.debug('Diagonalizing qeom matrices for excited states...')
+        a_mat = np.bmat([[m_mat, q_mat], [q_mat.T.conj(), m_mat.T.conj()]])
+        b_mat = np.bmat([[v_mat, w_mat], [-w_mat.T.conj(), -v_mat.T.conj()]])
+        # pylint: disable=too-many-function-args
+        res = linalg.eig(a_mat, b_mat)
+        # convert nan value into 0
+        res[0][np.where(np.isnan(res[0]))] = 0.0
+        # Only the positive eigenvalues are physical. We need to take care
+        # though of very small values
+        # should an excited state approach ground state. Here the small values
+        # may be both negative or
+        # positive. We should take just one of these pairs as zero. So to get the values we want we
+        # sort the real parts and then take the upper half of the sorted values.
+        # Since we may now have
+        # small values (positive or negative) take the absolute and then threshold zero.
+        logger.debug('... %s', res[0])
+        w = np.sort(np.real(res[0]))
+        logger.debug('Sorted real parts %s', w)
+        w = np.abs(w[len(w) // 2:])
+        w[w < 1e-06] = 0
+        excitation_energies_gap = w
+
+        return excitation_energies_gap, res[1]
+
+
+class QEOMResult(AlgorithmResult):
```

Comment:
So the excited states eigensolver end up I guess with an electronic structure result which hopefully has all the excited states printed out and formatted to as was done with the hamiltonian in the past.

This result here I guess embeds the lower level result, so its still available but has no nice formatted output

Reply:
we have this print in the electronic structure result, I am going to add it to the vibronic structure result